sdk/src/message_interface/message/account_method.rs

```diff
@@ -0,0 +1,472 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Serialize};
+
+#[cfg(feature = "participation")]
+use crate::wallet::account::types::participation::ParticipationEventRegistrationOptions;
+use crate::{
+    client::api::{PreparedTransactionDataDto, SignedTransactionDataDto},
+    types::block::{
+        dto::U256Dto,
+        output::{
+            dto::{AliasIdDto, NativeTokenDto, NftIdDto, OutputDto, TokenIdDto, TokenSchemeDto},
+            feature::dto::FeatureDto,
+            unlock_condition::dto::UnlockConditionDto,
+            FoundryId, OutputId,
+        },
+        payload::transaction::TransactionId,
+    },
+    wallet::{
+        account::{
+            handle::FilterOptions,
+            operations::{
+                address_generation::AddressGenerationOptions,
+                output_claiming::OutputsToClaim,
+                syncing::SyncOptions,
+                transaction::{
+                    high_level::{
+                        create_alias::AliasOutputOptionsDto,
+                        minting::{
+                            increase_native_token_supply::IncreaseNativeTokenSupplyOptionsDto,
+                            mint_native_token::NativeTokenOptionsDto, mint_nfts::NftOptionsDto,
+                        },
+                    },
+                    prepare_output::OutputOptionsDto,
+                    TransactionOptionsDto,
+                },
+            },
+        },
+        message_interface::dtos::{AddressWithAmountDto, AddressWithMicroAmountDto},
+        AddressAndNftId, AddressNativeTokens,
+    },
+};
+#[cfg(feature = "participation")]
+use crate::{
+    client::node_manager::node::Node,
+    types::api::plugins::participation::types::{ParticipationEventId, ParticipationEventType},
+};
+
+/// Each public account method.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(tag = "name", content = "data", rename_all = "camelCase")]
+pub enum AccountMethod {
```

Comment:
you are renaming all to camel-case, why are you also renaming individually?

Reply:
The rename_all only works for the variant names, not for fields of struct variants

Reply:
that's a bummer

Reply:
so are you saying that putting serde(rename_all) on the variants as well doesn't rename the fields?

Reply:
I just checked ... if you put the attribute on the variants the fields are correctly renamed.

Reply:
You're right, but still needs to be done for every variant that has such fields, issue for that https://github.com/iotaledger/iota-sdk/issues/192